src/common/utils/jsonp-helper.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+import {Env} from 'playkit-js'
+
+/**
+ * jsonp callback function, returns the direct manifest uri
+ * @param {Object} data - the json object that returns from the server
+ * @param {string} uri - original request uri
+ * @returns {string} returns the direct uri
+ */
+function getDirectManfiestUri(data: Object, uri: string): string {
+  const getParsedUri = uri => {
+    const parser = document.createElement('a');
+    parser.href = uri;
+    return {
+      'hostname': parser.hostname,
+      'uri': uri
+    }
+  };
+
+  const uriHost = getParsedUri(uri).hostname;
+  const flavorUriHost = getParsedUri(data.flavors[0].url).hostname
+  if (data.flavors.length === 1 && uriHost !== flavorUriHost) {
```

Comment:
@odedhutzler please document this part inline - since this logic is not very clear at first

Reply:
k